Lucid/Utils/Async.swift

```diff
@@ -0,0 +1,232 @@
+//
+//  Async.swift
+//  Lucid
+//
+//  Created by Ezequiel Munz on 14/04/2023.
+//  Copyright © 2023 Scribd. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+// MARK: - Async Tasks Storage
+
+public final actor AsyncTasks {
+
+    private var tasks: [Task<Void, any Error>] = []
+
+    deinit {
+        for task in tasks {
+            task.cancel()
+        }
+    }
+
+    public init() { }
+
+    public func cancel() {
+        for task in tasks {
+            task.cancel()
+        }
+    }
+}
+
+fileprivate extension AsyncTasks {
+
+    func append(_ task: Task<Void, any Error>) {
+        tasks.append(task)
+    }
+}
+
+public extension Task where Success == Void, Failure == any Error {
+
+    @discardableResult
+    func store(in asyncTasks: AsyncTasks) -> Self {
+        Task {
+            await asyncTasks.append(self)
+        }
+        return self
+    }
+}
+
+// MARK: - AsyncTasksQueue
+
+public actor AsyncTaskQueue {
+
+    private let concurrency: Int
+    private var running: Int = 0
+    private var queue = [CheckedContinuation<Void, Error>]()
+
+    public init(concurrency: Int = 2) {
```

Comment:
Why is the default 2 and not 1?

Also, what does `concurrency` mean? Would `maxConcurrentTasks` be more descriptive?